fzf: 0.47.0 -> 0.49.0

SebTM opened the pull request:
## Description of changes

* fzf: 0.47.0 -> 0.48.1
* Update derivation to not install old auto-completions
* Update module to load completion for bash, fish (new) and zsh (or oh-my-zsh plugin) with changed way through fzf-binary
* Added change to release note as it is backwards-incompatible.

Superseeds #295978

Home-Manager [module](https://github.com/nix-community/home-manager/blob/master/modules/programs/fzf.nix#L176) will need also an update (just as note here)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
fzf: 0.47.0 -> 0.48.1

SebTM committed:
fzf: Update package and module (shell-completions)

Update derivation to not install old shell-completions
Update module to load completion for bash, fish (new) and zsh (or oh-my-zsh plugin) with changed way through fzf-binary
Added change to release note as it is backwards-incompatible.

bjornfor commented on nixos/doc/manual/release-notes/rl-2405.section.md:
```diff
@@ -246,6 +246,8 @@ The pre-existing [services.ankisyncd](#opt-services.ankisyncd.enable) has been m
 
 - `xxd` has been moved from `vim` default output to its own output to reduce closure size. The canonical way to reference it across all platforms is `unixtools.xxd`.
 
+- `programs.fzf.keybindings` and `programs.fzf.fuzzyCompletion` got replaced by `programs.fzf.enabled` as shell-completion is included in the fzf-binary now there is no easy option to load completion and keybindings seperatly. Please consult fzf-documentation on how to configure/disable certain keybindings.
```
seperatly -> separately

bjornfor commented:
Please split the 236-char commit message line into a subject (<=50 chars) + blank line + body (wrapped to <= 72 chars).

SebTM commented on nixos/doc/manual/release-notes/rl-2305.section.md:
```diff
@@ -77,7 +77,7 @@ In addition to numerous new and updated packages, this release has the following
 - [frigate](https://frigate.video), an open source NVR built around real-time AI object detection. Available as [services.frigate](#opt-services.frigate.enable).
 
-- [fzf](https://github.com/junegunn/fzf), a command line fuzzyfinder. Available as [programs.fzf](#opt-programs.fzf.fuzzyCompletion).
+- [fzf](https://github.com/junegunn/fzf), a command line fuzzyfinder. Available as [programs.fzf](#opt-programs.fzf.enable).
```
Not sure if that's ok but the manual seems to build against the current state not the state it was made so I get a build-error of the changelog if the old option is still referenced with #opt here. Alternative would be to remove the link "#opt" and leave it as text.

SebTM commented:
> Please split the 236-char commit message line into a subject (<=50 chars) + blank line + body (wrapped to <= 72 chars).

Thanks for the feedback, all done 🙏🏻

SebTM commented:
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>50 packages built:</summary>
  <ul>
    <li>adl</li>
    <li>ani-cli</li>
    <li>arsenal</li>
    <li>arsenal.dist</li>
    <li>catcli</li>
    <li>catcli.dist</li>
    <li>clerk</li>
    <li>felix-fm</li>
    <li>fishPlugins.fzf-fish</li>
    <li>fontpreview</li>
    <li>fzf</li>
    <li>fzf-git-sh</li>
    <li>fzf-zsh</li>
    <li>fzf.man</li>
    <li>kakounePlugins.fzf-kak</li>
    <li>kns</li>
    <li>license-cli</li>
    <li>lunarvim</li>
    <li>mov-cli</li>
    <li>mov-cli.dist</li>
    <li>navi</li>
    <li>ocamlPackages.fzf</li>
    <li>ocamlPackages.magic-trace</li>
    <li>python311Packages.pyfzf</li>
    <li>python311Packages.pyfzf.dist</li>
    <li>python312Packages.pyfzf</li>
    <li>python312Packages.pyfzf.dist</li>
    <li>ripgrep-all</li>
    <li>spacevim</li>
    <li>sway-launcher-desktop</li>
    <li>sysz</li>
    <li>tmuxPlugins.extrakto</li>
    <li>tmuxPlugins.fuzzback</li>
    <li>tmuxPlugins.session-wizard</li>
    <li>tmuxPlugins.t-smart-tmux-session-manager</li>
    <li>tmuxPlugins.tmux-fzf</li>
    <li>unipicker</li>
    <li>vimPlugins.fzf-hoogle-vim</li>
    <li>vimPlugins.fzf-lua</li>
    <li>vimPlugins.fzf-vim</li>
    <li>vimPlugins.fzfWrapper</li>
    <li>vimPlugins.telescope-zoxide</li>
    <li>vimPlugins.vim-fzf-coauthorship</li>
    <li>vimPlugins.vim-zettel</li>
    <li>vimPlugins.zoxide-vim</li>
    <li>xmloscopy</li>
    <li>yazi</li>
    <li>ytfzf</li>
    <li>zoxide</li>
    <li>zsh-forgit</li>
  </ul>
</details>

SebTM commented:
Fixes issue where `systemPackages` was accidentally not a list ✌🏻

marsam commented:
Would you mind updating it to [fzf 0.49.0](https://github.com/junegunn/fzf/releases/tag/0.49.0)?

marsam committed:
fzf: remove fzf-share helper

The fzf-share helper was added to help to find the shell scripts [1],
but since fzf≥0.48.0 the it's no longer needed because fzf has its shell
integration scripts embedded.

[1] https://github.com/NixOS/nixpkgs/pull/27709

marsam committed:
fzf: remove perl dependency

Since fzf 0.43.0, the history command only uses Perl if it's installed
otherwise uses Awk [1].

[1] https://github.com/junegunn/fzf/commit/9f7684f6fe7a7889d1ebaded03b446c007d605ea

marsam committed:
fzf: 0.48.1 -> 0.49.0

Diff: https://github.com/junegunn/fzf/compare/0.48.1...0.49.0

Changelog: https://github.com/junegunn/fzf/blob/0.49.0/CHANGELOG.md

marsam committed:
nixos/fzf: fix fish shell commmand

marsam committed:
doc: replace fzf-share mention with sk-share from Skim package

`fzf-share` was removed

SebTM commented on nixos/modules/programs/fzf.nix:
```diff
@@ -30,7 +30,7 @@ in
 
     programs.fish.interactiveShellInit = ''
-      eval "$(${getExe pkgs.fzf} --fish)"
+      ${getExe pkgs.fzf} --fish | source
```
Should we do this for the others as well?

marsam replied:
I think it's just the [`fish` setup that is different](https://github.com/junegunn/fzf?tab=readme-ov-file#setting-up-shell-integration)
<img width="437" alt="image" src="https://github.com/NixOS/nixpkgs/assets/65531/19cfd446-9667-45eb-a80a-a2e9bca39e9b">

SebTM commented:
Thanks for updating, I guess we need another round of review now before we can merge 🙏🏻 
Will have a look soonTM ✌🏻

marsam commented:
I'm going ahead and merge this, because another `fzf` bump was merged a few hours ago https://github.com/NixOS/nixpkgs/pull/301585, and it's likely to cause errors with the fzf module.

SuperSandro2000 commented:
> Update derivation to not install old auto-completions

The shell scripts are just embedded (https://github.com/junegunn/fzf/blob/master/main.go#L15-L28) using go's inbuilt embed feature.
So we lost options here for nothing and we should probably revert this to easily disable completion or keybindings again.

SuperSandro2000 commented:
We can just add an option for that https://github.com/junegunn/fzf/pull/3722 :)

#303213

SuperSandro2000 commented:
I've changed #303213 according to upstream feedback and half of the changes done here are being reverted because there is no reason for them and upstream recommend to load the shell files individually, if you want more control, so it seems like there is no urge to switch to the new format which makes removing options unjustified. Please take a look at that PR asap.
